### ISSUE TITLE
Microsoft R Host always stop working

### ISSUE THREAD

**Issue by Shidiq** [Microsoft/RTVS#4307]
Please briefly describe what you were doing that led to the issue, if applicable:
**Microsoft R Host always stop working**

![image](https://user-images.githubusercontent.com/23032122/38448121-3261c4ac-3a2c-11e8-9c46-4dc485f281ba.png)

![image](https://user-images.githubusercontent.com/23032122/38448115-24e0ac62-3a2c-11e8-9fc2-b290a6cc4190.png)

**Details:**
Source
Microsoft R Host

Summary
Stopped working

Date
‎4/‎7/‎2018 6:15 AM

Status
Ready for upload

Description
Faulting Application Path:	C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\Extensions\Microsoft\R Tools for Visual Studio\Microsoft.R.Host.exe

Problem signature
Problem Event Name:	APPCRASH
Application Name:	Microsoft.R.Host.exe
Application Version:	1.3.31012.1601
Application Timestamp:	5a4ea255
Fault Module Name:	ntdll.dll
Fault Module Version:	10.0.17133.1
Fault Module Timestamp:	e49d47fd
Exception Code:	c00000fd
Exception Offset:	0000000000035d91
OS Version:	10.0.17133.2.0.0.256.121
Locale ID:	1033
Additional Information 1:	0c5f
Additional Information 2:	0c5fa48a403d3d14fa77ebf43cc5eb4b
Additional Information 3:	0b9f
Additional Information 4:	0b9f07d35cb90b91681d7231da9f2116

Files that help describe the problem
minidump.mdmp
WERInternalMetadata.xml
WERInternalRequest.xml
memory.csv
sysinfo.txt
AppCompat.txt
WERDataCollectionStatus.txt
memory.hdmp

...

If you have any screenshots demonstrating the issue, please include them as well to help us diagnose it better.

Additional information:

OS Information
    Version:       Microsoft Windows NT 10.0.17133.0

RTVS Information:
    Assembly: Microsoft.VisualStudio.R.Package, Version=1.3.40104.1351, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a

Active R URI:
    Microsoft R Open (3.4.3.0): C:\Program Files\Microsoft\R Open\R-3.4.3\

**Comment by smortaz** [comment]
Hi.  Can you please send us the RTVS logs?  It’s in the RTVS menu. Thx

**Comment by Shidiq** [comment]
@smortaz here my RTVS logs. Thanks.

[RTVS_2018407_061318_pid12688.log](https://github.com/Microsoft/RTVS/files/1885827/RTVS_2018407_061318_pid12688.log)

**Comment by MikhailArkhipov** [comment]
The error code is 'stack overflow'. Does code work in RGUI?

**Comment by Shidiq** [comment]
@MikhailArkhipov yes, the code work well in RGUI and RStudio witout any errors. I try using Visual Studio, but from the first open R in visual studio, R Host Stop working when I write code.

**Comment by Nekochef** [comment]
I have had similar issue for a long time. After VS was opened and project was loaded, the CPU usage of Microsoft R Host would suddenly climb up to about 30%, though I didn't do any thing. Then it seemed that the R Host is crashed, and the windows problem reporting process appeared, and finally disappeared.
That phenomenon happened repeatedly, but I can use RTVS to run code blocks without a problem. I have no clue with this issue.

**Comment by Nekochef** [comment]
**UPDATE：**
I solved this issue by checking Run as administrator and Run in win7 compatibility mode in Microsoft.R.Host.exe 's properties page:
![image](https://user-images.githubusercontent.com/22209183/38570354-3ae2abf4-3d20-11e8-9078-4fe9155ca12b.png)
After doing this, the second instance of Microsoft.R.Host.exe never appears and crashes again and again.
Any idea?

**Comment by Shidiq** [comment]
@Nekochef unfortunately, it's not working on me.

![2018-04-11_061635](https://user-images.githubusercontent.com/23032122/38588753-bfbcd4ae-3d51-11e8-8f61-e99c24d6b117.png)


[RTVSLogs.zip](https://github.com/Microsoft/RTVS/files/1896751/RTVSLogs.zip)

**Comment by Nekochef** [comment]
Hi @Shidiq , have you tried to change the default library paths?

Just create a new folder for your library, and find **Rprofile.site** in %R directory%\etc\, then add
`.libPaths("Your Folder Path")` in a new line. You may need to run your text editor as administrator. This will change your default library path, thus you'll have to reinstall all the packages needed, but for me, this time it really solved my issue. When taking no action, VS and RTVS won't use any CPU or memory.

I think it is due to some incompatible packages, because since I first use R, I didn't change my library path, there might be too many packages that can affect each other in some ways, or at least incompatible to RTVS. I even reinstalled my system, then reinstalled VS 2017 but the crashes keep happening, then I realise that the only thing remained was my library path.

Hope it can help.

**Comment by Shidiq** [comment]
Hi @Nekochef thank you very much. It works great on me. CPU stable and no error again.

![2018-04-11_155727](https://user-images.githubusercontent.com/23032122/38606776-2ca8e734-3da1-11e8-82e3-f73f7364c7fb.png)
 
Thank you.

**Comment by MikhailArkhipov** [comment]
Will look into this for 15.7 P5. Looks like intellisense process is recursing and crashing.

**Comment by priscian** [comment]
Incidentally, is it possible in Workspaces to set the library path through "R command line arguments"? From PowerShell, I can do:

`R[gui].exe --vanilla R_LIBS_USER=C:/Users/jim/Documents/R/library/devel`

to put my non-R_HOME library first on the search path inside the R session, but RTVS doesn't seem to allow command-line assignment of environment vars, which makes it difficult to use personal libraries. (`--vanilla` above seems to work, however, with RTVS.)

**Comment by priscian** [comment]
I just upgraded to VS Community 15.8.7 and checked out RTVS—it appears that this issue still exists. I'm hoping that resolution is still on the docket. Cheers.

**Comment by priscian** [comment]
Checking in again. Are there any plans to resolve this problem?
